Subject: [PATCH] #36: Added navigation 4
---
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/Screen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/Screen.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/Screen.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/Screen.kt	(revision 3438bb9b6132168a986e4502e208006de5340e99)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/Screen.kt	(date 1711276603624)
@@ -9,6 +9,7 @@
     SPORT_COURT_INFO_SCREEN("sport_court_info_screen"),
     TRAINING_LIST_SCREEN("training_list_screen"),
     TRAINING_CREATE_SCREEN("training_create_screen"),
+    RUNNING_LIST_SCREEN("running_list_screen"),
     SETTINGS_MAIN_SCREEN("settings_main_screen"),
     SETTINGS_UPDATE_PROFILE_SCREEN("settings_update_profile_screen")
 }
@@ -18,5 +19,6 @@
     PROFILE("profile"),
     SPORT_COURT("sport_court"),
     TRAINING("training"),
+    RUNNING("running"),
     SETTINGS("settings")
 }
\ No newline at end of file
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/NavigationController.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/NavigationController.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/NavigationController.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/NavigationController.kt	(revision 3438bb9b6132168a986e4502e208006de5340e99)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/NavigationController.kt	(date 1711276603625)
@@ -51,4 +51,10 @@
 
     fun navigateToTrainingCreate() =
         navHostController.navigate(Screen.TRAINING_CREATE_SCREEN.route)
+
+    fun navigateToRunning() =
+        navHostController.navigate(ScreensSubgraphs.RUNNING.route)
+
+    fun navigateToRunningList() =
+        navHostController.navigate(Screen.RUNNING_LIST_SCREEN.route)
 }
\ No newline at end of file
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/MainNavHost.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/MainNavHost.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/MainNavHost.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/MainNavHost.kt	(revision 3438bb9b6132168a986e4502e208006de5340e99)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/navigation/MainNavHost.kt	(date 1711276603630)
@@ -22,6 +22,8 @@
 import ru.moevm.sportfinder.screen.common_components.common_top_bar.CommonTopBarTypeBuilder
 import ru.moevm.sportfinder.screen.profile.ProfileScreen
 import ru.moevm.sportfinder.screen.profile.ProfileViewModel
+import ru.moevm.sportfinder.screen.running.RunningListScreen
+import ru.moevm.sportfinder.screen.running.RunningListViewModel
 import ru.moevm.sportfinder.screen.settings.MainSettingsScreen
 import ru.moevm.sportfinder.screen.settings.SettingsViewModel
 import ru.moevm.sportfinder.screen.settings.UpdateProfileSettingsScreen
@@ -212,6 +214,26 @@
             }
         }
 
+        navigation(
+            startDestination = Screen.RUNNING_LIST_SCREEN.route,
+            route = ScreensSubgraphs.RUNNING.route
+        ) {
+            composable(route = Screen.RUNNING_LIST_SCREEN.route) {
+                val viewModel = hiltViewModel<RunningListViewModel>()
+                val state by viewModel.state.collectAsStateWithLifecycle()
+                updateBottomBarVisible(true)
+                updateTopBarType(false, null)
+
+                RunningListScreen(
+                    state = state,
+                    onTextForFilterChanged = viewModel::onTextForFilterChanged,
+                    onFilterApply = viewModel::onFilterApply,
+                    navigateToRunningInfoScreen = { /*TODO*/ },
+                    navigateToRunningCreateScreen = { /*TODO*/ }
+                )
+            }
+        }
+
         navigation(
             startDestination = Screen.SETTINGS_MAIN_SCREEN.route,
             route = ScreensSubgraphs.SETTINGS.route
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/MainActivity.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/MainActivity.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/MainActivity.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/MainActivity.kt	(revision 3438bb9b6132168a986e4502e208006de5340e99)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/MainActivity.kt	(date 1711276603636)
@@ -90,7 +90,7 @@
         ),
         BottomNavItem(
             R.drawable.ic_bottom_bar_running_track,
-            {  }
+            navigationController::navigateToRunning
         ),
         BottomNavItem(
             R.drawable.ic_bottom_bar_training,
