Subject: [PATCH] #62: Ui fixes
---
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingListScreen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingListScreen.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingListScreen.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingListScreen.kt	(revision 30f80f9f9e15c40e6ee54422e8cb88b757fb3e31)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingListScreen.kt	(date 1711543663789)
@@ -24,6 +24,7 @@
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
@@ -56,12 +57,12 @@
                 verticalArrangement = Arrangement.spacedBy(4.dp),
             ) {
                 if (isLoading) {
-                    items(3) {
+                    items(8) {
                         Box(
                             modifier = Modifier
                                 .height(100.dp)
                                 .fillMaxWidth()
-                                .padding(10.dp)
+                                .clip(RoundedCornerShape(5))
                                 .shimmerEffect()
                         )
                     }
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingCreateViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingCreateViewModel.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingCreateViewModel.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingCreateViewModel.kt	(revision 30f80f9f9e15c40e6ee54422e8cb88b757fb3e31)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/training/TrainingCreateViewModel.kt	(date 1711544482422)
@@ -50,6 +50,9 @@
     }
 
     fun onSaveClick() {
+        if (_state.value.name.isBlank() || _state.value.description.isBlank()) {
+            return
+        }
         // TODO: Сделать сохранение 
     }
 }
\ No newline at end of file
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningCreateViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningCreateViewModel.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningCreateViewModel.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningCreateViewModel.kt	(revision 30f80f9f9e15c40e6ee54422e8cb88b757fb3e31)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningCreateViewModel.kt	(date 1711543961903)
@@ -61,6 +61,9 @@
     }
 
     fun onSaveClick(onSuccess: () -> Unit) {
+        if (_state.value.title.isBlank() || _state.value.listOfPoints.isEmpty()) {
+            return
+        }
         useRunningDatabaseUseCase.addRunning(
             _state.value.title,
             _state.value.listOfTags,
Index: SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningListScreen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningListScreen.kt b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningListScreen.kt
--- a/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningListScreen.kt	(revision 30f80f9f9e15c40e6ee54422e8cb88b757fb3e31)
+++ b/SportFinder/app/src/main/java/ru/moevm/sportfinder/screen/running/RunningListScreen.kt	(date 1711543663793)
@@ -24,6 +24,7 @@
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
@@ -56,12 +57,12 @@
                 verticalArrangement = Arrangement.spacedBy(4.dp),
             ) {
                 if (isLoading) {
-                    items(3) {
+                    items(8) {
                         Box(
                             modifier = Modifier
                                 .height(100.dp)
                                 .fillMaxWidth()
-                                .padding(10.dp)
+                                .clip(RoundedCornerShape(5))
                                 .shimmerEffect()
                         )
                     }
